requests/utils.py

```diff
@@ -726,3 +726,11 @@ def urldefragauth(url):
     netloc = netloc.rsplit('@', 1)[-1]
 
     return urlunparse((scheme, netloc, path, params, query, ''))
+
+
+def determine_if_stream(data):
+    """Given data, determines if it should be sent as a stream.
+    """
+    is_iterable = hasattr(data, '__iter__')
```

Comment:
Probably doing the `__getattr__` dance from @hynek's blog post is a good idea here, thanks @ueg1990.